frontend/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,97 @@
+import { useRef } from 'react'
+import { BiMinus, BiPlus } from 'react-icons/bi'
+import {
+  Box,
+  chakra,
+  Divider,
+  forwardRef,
+  NumberInputProps as ChakraNumberInputProps,
+  useFormControlProps,
+  useMultiStyleConfig,
+  useNumberInput,
+} from '@chakra-ui/react'
+
+import IconButton from '../IconButton'
+
+export interface NumberInputProps extends ChakraNumberInputProps {
+  /**
+   * Whether the input is in a prefilled state.
+   */
+  isPrefilled?: boolean
+  /**
+   * Whether the input is in a success state.
+   */
+  isSuccess?: boolean
+  /**
+   * Whether to show the increment and decrement steppers. Defaults to true.
+   */
+  showSteppers?: boolean
+}
+
+export const NumberInput = forwardRef<NumberInputProps, 'div'>(
+  (
+    {
+      showSteppers = true,
+      clampValueOnBlur = false,
+      isSuccess,
+      isPrefilled,
+      ...props
+    },
+    ref,
+  ) => {
+    const styles = useMultiStyleConfig('NumberInput', {
+      ...props,
+      isSuccess,
+      isPrefilled,
+    })
+
+    const stepperWrapperRef = useRef<HTMLDivElement | null>(null)
+
+    const controlProps = useFormControlProps(props)
+    const {
+      htmlProps,
+      getInputProps,
+      getIncrementButtonProps,
+      getDecrementButtonProps,
+    } = useNumberInput({ ...controlProps, clampValueOnBlur })
```

Comment:
I find it EXTREMELY frustrating that `chakra` doesn't have any documentation on this. how do you know to use `useFormControlProps`? how dyou know what args it takes, and that the return value can be passed back into `useNumberInput`?

Reply:
I looked at the source code for their own NumberInput implementation.

Reason for using `useFormControlProps` is so that this component can be wrapped with a `FormControl` component and be able to access their relevant props, such as whether the field is disabled, readOnly, invalid, etc.

Reply:
Possible to patch upstream, i will open an issue if I have time....